src/stores/schedule.ts

```diff
@@ -0,0 +1,43 @@
+//The below types are derived from Algorithm 1's specifications
+//For now, course, professor and the dates are using easier types to work with, just to assist with mock data.
+//In the future remove those though
+
+export type Assignment = {
+  course: Course | string;
+  professor: Professor | string;
+  startDate: string | Date; // Follow "yyyy-mm-dd"
+  endDate: string | Date; // Follow "yyyy-mm-dd"
+  beginTime: string; // Use 24hr "0000" - "2359"
+  endTime: string; // Use 24hr "0000" - "2359"
+  monday?: boolean;
+  tuesday?: boolean;
+  wednesday?: boolean;
+  thursday?: boolean;
+  friday?: boolean;
```

Comment:
This is a bit of an odd way to represent the days, but I'm guessing it's how the algorithm 1 team wrote it?

Reply:
yeah just basing it off of algo 1's spec